### PR TITLE
[Upgrade Tool] Allow .json transforms

### PR DESCRIPTION
**Pull request by Convly**
### What does it do?

- Allow running both `json` and `code` transform tasks
- Upgrade glob to the latest version to beneficiate from the new features (also update its usage in strapi/strapi)

### Why is it needed?

Since jscodeshift doesn't parse `json` files, we had to find a way of running transformations in those files (versions upgrade, etc...)

### How to test it?

- Run the upgrade tool (dry mode), the examples code and json transforms should be picked up

### CODE REVIEW

File: packages/utils/upgrade/src/core/project-loader.ts [hunk 1]
```diff
@@ -81,19 +82,19 @@ const loadProjectFiles = async (options: ProjectLoaderOptions): Promise<string[]
...
+  const patterns = [projectFilesPattern, ...PROJECT_DEFAULT_PATTERNS];
 
-  const files = await new Promise<string[]>((resolve, reject) => {
-    glob(pattern, { cwd }, (err, matches) => (err ? reject(err) : resolve(matches)));
-  });
+  const files = await glob(patterns, { cwd });
 
   const fFilesLength = f.highlight(files.length.toString());
-  const fPattern = f.highlight(pattern);
+  const fPattern = f.highlight(patterns.map((p) => `"${p}"`).join(', '));
```
Comment (innerdvations) [inline packages/utils/upgrade/src/core/project-loader.ts:91]: `"` is a valid character in paths, so we may need to escape it (or tbh if it's more complicated than expected we could just document that the allowedRootPaths must be an escaped value, it's not a common use case)

**Review by innerdvations — APPROVED** [review]
Looks great, only minor suggestions but nothing required.

File: packages/core/strapi/package.json [hunk 2]
```diff
@@ -142,7 +142,7 @@
     "dotenv": "14.2.0",
     "execa": "5.1.1",
     "fs-extra": "10.0.0",
-    "glob": "7.2.3",
+    "glob": "10.3.10",
```
[the remark this answers is not in the document]
Comment (Convly) [reply packages/core/strapi/package.json:145]: From the investigation I've done, we're only using glob in strapi/strapi `loadFiles`, which I have updated & tested manually.
v10 brings a lot of features that we kinda need in the upgrade tool, I'm not sure what's the best way to proceed with this.

File: packages/utils/upgrade/src/core/format.ts [hunk 3]
```diff
@@ -51,29 +51,7 @@ export const reports = (reports: RunReports) => {
...
-
-  if (restMs >= ONE_SECOND_MS) {
-    concat(Math.floor(restMs / ONE_SECOND_MS), 's');
-    restMs %= ONE_SECOND_MS;
-  }
-
-  concat(restMs, 'ms');
-
-  return str;
+  return `${elapsedSeconds}s`;
```
[the remark this answers is not in the document]
Comment (Convly) [reply packages/utils/upgrade/src/core/format.ts:56]: I agree, here I'm using seconds to match jscodeshift elapsed time formatting, but we can change that in the future

File: packages/utils/upgrade/src/core/project-loader.ts (hunk 1 above)
Comment (Convly) [reply packages/utils/upgrade/src/core/project-loader.ts:91]: fPattern is only for logging purpose though, does it still make sense to take the escaping in consideration?

**Pull request merged**